### PR TITLE
added test for GeneralEditor component

### PR DESCRIPTION
**Pull request by NabokinAlexandr**
develop

## GitHub Board

**Issue link**

[#11 Issue name ]( )

**Story link**

[#11 Story]( )

## Code reviewers
- [ ] @github_username

## Summary of issue

- [ ]  Summary of issue
- [ ]  Summary of issue

## Summary of change

- [ ]  Summary of change
- [ ]  Summary of change

### CODE REVIEW

File: src/components/Editor/GeneralEditor.test.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import { GeneralEditor } from './GeneralEditor';
+import { VideoEditorToolbar } from './Editors/VideoEditorToolbar';
+import { JSDOM } from 'jsdom';
```
Comment (michalenr) [inline src/components/Editor/GeneralEditor.test.tsx:5]: there is no need to hook to jsdom directly. Jest uses jsdom under the hood

Comment (NabokinAlexandr) [reply src/components/Editor/GeneralEditor.test.tsx]: done

**Review by supersoild2000 — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -12,6 +12,7 @@
     "dompurify": "^2.2.9",
     "history": "^4.9.0",
     "i18next": "^20.3.1",
+    "jsdom": "^16.6.0",
```
Comment (michalenr) [inline package.json:15]: you can remove it from dependencies

File: src/components/Editor/GeneralEditor.test.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-this-alias */
+
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import { GeneralEditor } from './GeneralEditor';
+import { VideoEditorToolbar } from './Editors/VideoEditorToolbar';
+
+global.document.execCommand = jest.fn((value) => value);
```
Comment (michalenr) [inline src/components/Editor/GeneralEditor.test.tsx:12]: what is this for?

File: src/components/Editor/GeneralEditor.test.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+const mockHtmlContentCange = jest.fn((value) => value);
```
Comment (michalenr) [inline src/components/Editor/GeneralEditor.test.tsx:14]: you can replace all of these with `jest.fn()`

File: src/components/Editor/GeneralEditor.test.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+const mockTextContantChange = jest.fn((value) => value);
+jest.mock(
+  'quill-image-uploader',
+  () =>
+    function (quill, options) {
+      const self = this;
```
Comment (michalenr) [inline src/components/Editor/GeneralEditor.test.tsx:20]: why is this needed? 

Here's how to mock classes wiht jest

https://jestjs.io/docs/es6-class-mocks

File: src/components/Editor/GeneralEditor.test.tsx (hunk 3 above)
Comment (NabokinAlexandr) [reply src/components/Editor/GeneralEditor.test.tsx:12]: > what is this for?

the test can't run because execCommand (it's a part of library) is not a function. So I mocked it.

File: src/components/Editor/GeneralEditor.test.tsx (hunk 5 above)
Comment (NabokinAlexandr) [reply src/components/Editor/GeneralEditor.test.tsx:20]: imageUploaderError

Comment (NabokinAlexandr) [reply src/components/Editor/GeneralEditor.test.tsx:20]: imageUploaderSuccess

File: package.json (hunk 2 above)
Comment (NabokinAlexandr) [reply package.json:15]: done

**Review by michalenr — APPROVED** [review]
